h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1263,18 +1279,40 @@ public int nclasses() {
       return res;
     }
 
-    public Submodel pickBestModel() {
-      int bestId = 0;
-      Submodel best = _submodels[0];
-      for(int i = 1; i < _submodels.length; ++i) {
-        Submodel sm = _submodels[i];
-        if(!(sm.devianceTest > best.devianceTest) && sm.devianceTrain < best.devianceTrain){
-          bestId = i;
-          best = sm;
+    public double selectBestSubmodel(double trainDev, double testDev, double[] xval_test_deviances, int alphaIndex,
+                                     int lambdaIndex, int submodelIdx, double bestDevValue, boolean setAlphaLambdaIdx) {
+      // if testDev exists, use it above everything else;
```

Comment:
by testDev, you mean validationDev?
test data should not be used in the selection process.
But if you mean validation, then usually xval takes precedence over validation, at least that's what we do with metrics.

Reply:
@sebhrusen Here is the deal, during xval, the training frame is split into training folds and validation folds.  The testDev will refer to the results from the validation.  During the final main model training, if the lambdaCVestimate is not null, the xval decision will be used to determine the best model.  So, your wish has come true in this case.

Reply:
ok, can you please rename to `validationDev` then to avoid confusion?

Reply:
I removed the function but did try to rename the other test deviance to valid deviance.